addons/pos_restaurant/static/src/js/floors.js

```diff
@@ -1,97 +1,150 @@
 odoo.define('pos_restaurant.floors', function (require) {
 "use strict";
 
-var { PosGlobalState, Order, Payment } = require('point_of_sale.models');
-const { Gui } = require('point_of_sale.Gui');
+const { PosGlobalState, Order, Orderline, Payment } = require('point_of_sale.models');
 const Registries = require('point_of_sale.Registries');
 
+const TIMEOUT = 7500;
 
 // New orders are now associated with the current table, if any.
 const PosRestaurantOrder = (Order) => class PosRestaurantOrder extends Order {
     constructor(obj, options) {
         super(...arguments);
         if (this.pos.config.module_pos_restaurant) {
-            if (this.pos.config.iface_floorplan && !this.table && !options.json) {
-                this.table = this.pos.table;
+            if (this.pos.config.iface_floorplan && !this.tableId && !options.json) {
+                this.tableId = this.pos.table.id;
             }
-            this.customer_count = this.customer_count || 1;
+            this.customerCount = this.customerCount || 1;
         }
     }
     export_as_JSON() {
         var json = super.export_as_JSON(...arguments);
         if (this.pos.config.module_pos_restaurant) {
             if (this.pos.config.iface_floorplan) {
-                json.table = this.table ? this.table.name : undefined;
-                json.table_id = this.table ? this.table.id : false;
-                json.floor = this.table ? this.table.floor.name : false;
-                json.floor_id = this.table ? this.table.floor.id : false;
+                json.table_id = this.tableId
+                json.floor = this.getTable().floor.name;
+                json.floor_id = this.getTable().floor.id;
             }
-            json.customer_count = this.customer_count;
+            json.customer_count = this.customerCount;
         }
         return json;
     }
     init_from_JSON(json) {
         super.init_from_JSON(...arguments);
         if (this.pos.config.module_pos_restaurant) {
             if (this.pos.config.iface_floorplan) {
-                this.table = this.pos.tables_by_id[json.table_id];
-                this.floor = this.table ? this.pos.floors_by_id[json.floor_id] : undefined;
+                this.tableId = json.table_id;
+                this.floor = this.pos.floors_by_id[json.floor_id];
```

Comment:
I checked and there is no use of `floor` object derived from the `order`. Can you confirm? Maybe we can remove this assignment, and just access the floor object thru the order's table.